lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -108,7 +118,12 @@
 <%# additional disks -%>
 <% @disks.each do |d| -%>
     <disk type='file' device='disk'>
-      <driver name='qemu' type='<%= d[:type] %>' cache='<%= d[:cache] %>'/>
+      <driver name='qemu' type='<%= d[:type] %>' cache='<%= d[:cache] %>'<%= -%>
+<%# Extra optional disk driver atrributes -%>
+<%=      " io='#{d[:io]}'"                           unless d[:io].nil? -%>
+<%=      " copy_on_read='#{d[:copy_on_read]}'"       unless d[:copy_on_read].nil? -%>
+<%=      " discard='#{d[:discard]}'"                 unless d[:discard].nil? -%>
+<%=      " detect_zeroes='#{d[:detect_zeroes]}'"     unless d[:detect_zeroes].nil? -%>/>
```

Comment:
Similarly is it worth just using a loop here with a defined list of params to read from?

```
<%
       keys = [:type, :cache, :io, :copy_on_read, :discard, :detect_zeros]
       @opts = d.select |key, value| { @keys.include? key && !value.nil? }.each |key, value|
            "#{key.to_str}=#{value}"
       end.join(' ')
-%>
      <driver name='qemu' <%= @opts -%> />
```